compiler/src/iree/compiler/GlobalOptimization/GlobalLoopInvariantCodeMotion.cpp

```diff
@@ -0,0 +1,148 @@
+// Copyright 2024 The IREE Authors
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "iree/compiler/GlobalOptimization/PassDetail.h"
+#include "iree/compiler/GlobalOptimization/Passes.h"
+#include "llvm/ADT/TypeSwitch.h"
+#include "mlir/Dialect/SCF/IR/SCF.h"
+#include "mlir/Dialect/SCF/Transforms/Transforms.h"
+#include "mlir/Dialect/Tensor/IR/Tensor.h"
+#include "mlir/IR/PatternMatch.h"
+#include "mlir/Transforms/TopologicalSortUtils.h"
+
+#include "llvm/Support/Debug.h"
+
+using namespace mlir;
+
+#define DEBUG_TYPE "global-loop-invariant-code-motion"
+#define LICM_DBGS() (llvm::dbgs() << '[' << DEBUG_TYPE << "] ")
+
+// Check if the op and its producers are loop invariants and hoistable. Results
+// are cached in hoistableOpMap.
+//
+// It expects the op's producers have been checked and the results exist in
+// `hoistableOpMap`.
+static bool
+isHoistableOp(LoopLikeOpInterface loopOp, Operation *op,
```

Comment:
You can actually dont need a map here. You just need a `SetVector`. If it is in the set it is hoistable, if it is not then its not hoistable. `Map<Operation *, bool>` is effectively a `Set`.

Reply:
Done.